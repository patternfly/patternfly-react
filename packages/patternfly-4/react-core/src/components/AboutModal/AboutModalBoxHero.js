import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/AboutModalBox/about-modal-box.css';

const propTypes = {
  /** additional classes added to the About Modal Hero */
  className: PropTypes.string,
  /** the URL of the image for the Hero. */
  src: PropTypes.string,
  /** the alternate text of the Hero image. */
  alt: PropTypes.string.isRequired
};

const defaultProps = {
  className: '',
  src: ''
};

const AboutModalBoxHero = ({ className, src, alt, ...props }) => (
  <div {...props} className={css(styles.aboutModalBoxHero, className)}>
    <img className={css(styles.aboutModalBoxHeroImage)} src={src} alt={alt} />
  </div>
);

AboutModalBoxHero.propTypes = propTypes;
AboutModalBoxHero.defaultProps = defaultProps;

export default AboutModalBoxHero;
