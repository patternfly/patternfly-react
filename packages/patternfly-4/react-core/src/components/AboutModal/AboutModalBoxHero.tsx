import * as React from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/AboutModalBox/about-modal-box';
import { c_about_modal_box__hero_sm_BackgroundImage } from '@patternfly/react-tokens'

export interface AboutModalBoxHeroProps extends React.HTMLProps<HTMLDivElement> {
  /** additional classes added to the About Modal Hero  */
  className?: string;
  /** background image data or file path  */
  backgroundImageSrc?: string;
}

export const AboutModalBoxHero: React.SFC<AboutModalBoxHeroProps> = ({
  className,
  backgroundImageSrc,
  ...props
}: AboutModalBoxHeroProps) => (
  <div
    style={backgroundImageSrc !== ''
      ? { [c_about_modal_box__hero_sm_BackgroundImage.name as string]: `url(${backgroundImageSrc})`}
      : {}}
    className={css(styles.aboutModalBoxHero, className)}
    {...props} />
);
