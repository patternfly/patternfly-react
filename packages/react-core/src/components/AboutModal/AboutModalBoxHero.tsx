import * as React from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/AboutModalBox/about-modal-box';
// eslint-disable-next-line camelcase
import c_about_modal_box__hero_sm_BackgroundImage from '@patternfly/react-tokens/dist/js/c_about_modal_box__hero_sm_BackgroundImage';

export interface AboutModalBoxHeroProps extends React.HTMLProps<HTMLDivElement> {
  /** additional classes added to the About Modal Hero  */
  className?: string;
  /** background image data or file path  */
  backgroundImageSrc?: string;
}

export const AboutModalBoxHero: React.FunctionComponent<AboutModalBoxHeroProps> = ({
  className,
  backgroundImageSrc,
  ...props
}: AboutModalBoxHeroProps) => (
  <div
    style={
      /* eslint-disable camelcase */
      backgroundImageSrc !== ''
        ? { [c_about_modal_box__hero_sm_BackgroundImage.name as string]: `url(${backgroundImageSrc})` }
        : {}
      /* eslint-enable camelcase */
    }
    className={css(styles.aboutModalBoxHero, className)}
    {...props}
  />
);
AboutModalBoxHero.displayName = 'AboutModalBoxHero';
