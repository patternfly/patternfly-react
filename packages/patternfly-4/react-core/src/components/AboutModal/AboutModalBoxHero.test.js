import React from 'react';
import { shallow } from 'enzyme';
import AboutModalBoxHero from './AboutModalBoxHero';

test('test About Modal Box SHero', () => {
  const view = shallow(<AboutModalBoxHero src="heroImg..." alt="hero" />);
  expect(view).toMatchSnapshot();
});
