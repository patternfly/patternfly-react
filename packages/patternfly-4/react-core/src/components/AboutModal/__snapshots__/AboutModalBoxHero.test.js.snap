// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test About Modal Box SHero 1`] = `
<div
  className="pf-c-about-modal-box__hero"
  style={
    Object {
      "--pf-c-about-modal-box__hero--sm--BackgroundImage": "url(undefined)",
    }
  }
/>
`;
